rclpy/rclpy/context.py

```diff
@@ -95,3 +108,12 @@ def on_shutdown(self, callback: Callable[[], None]):
                 callback()
             else:
                 self._callbacks.append(weakref.WeakMethod(callback, self._remove_callback))
+
+    def _logging_fini(self):
+        from rclpy.impl.implementation_singleton import rclpy_implementation
+        global g_logging_ref_count
+        if self._logging_initialized:
+            with g_logging_configure_lock:
+                g_logging_ref_count -= 1
+                if g_logging_ref_count == 0:
+                    rclpy_implementation.rclpy_logging_fini()
```

Comment:
@ivanpauno should `self._logging_initialized` go back to False?

Reply:
See https://github.com/ros2/rclpy/pull/518/commits/a5814ed5b7c5788ecd6e77aa11fa36580ed8efaa.